framework/framework.go

```diff
@@ -97,13 +98,43 @@ func (f *Framework) ProcessCreate(obj interface{}, resources ...Resource) error
 	return nil
 }
 
+// ProcessCreateWithBackoff is the same as ProcessCreate but takes an additional
+// backoff factory returning new backoff implementations to retry the creation
+// on resource errors. Retries are implemented using a retry resource that wraps
+// each given resource. Errors are logged eventually in case an appropriate
+// logger is configured in the operator framework.
+func (f *Framework) ProcessCreateWithBackoff(obj interface{}, resources []Resource, backoffFactory func() backoff.BackOff) error {
+	var retryResources []Resource
+
+	for _, r := range resources {
+		resourceConfig := DefaultRetryResourceConfig()
+		resourceConfig.BackOff = backoffFactory()
+		resourceConfig.Logger = f.logger
+		resourceConfig.Resource = r
+
+		retryResource, err := NewRetryResource(resourceConfig)
```

Comment:
> [backoff] applied in the Process* methods around resource methods invocations instead of creating a new resource type

This is the proposed solution.

> We could get rid of the retry resource if we wrap each and every method into its own backoff operation.

I think this is better. Then we don't have to use separate resource. Also `Process*` functions would be only invocations of `Process*BackOff` with a default backOff factory which feels a lot cleaner.

---

I leave it to you <3